src/main/java/uk/gov/indexer/dao/TotalRegisterEntriesUpdateDao.java

```diff
@@ -0,0 +1,16 @@
+package uk.gov.indexer.dao;
+
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+
+interface TotalRegisterEntriesUpdateDao {
+    String TOTAL_REGISTER_ENTRIES_TABLE = "register_entries_count";
+
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS " + TOTAL_REGISTER_ENTRIES_TABLE + " (COUNT INTEGER)")
+    void ensureTotalEntriesInRegisterTableExists();
+
+    @SqlUpdate("INSERT INTO " + TOTAL_REGISTER_ENTRIES_TABLE + "(COUNT) SELECT 0 WHERE NOT EXISTS (SELECT 1 FROM " + TOTAL_REGISTER_ENTRIES_TABLE + ")")
+    void initialiseTotalEntriesInRegisterIfRequired();
+
+    @SqlUpdate("UPDATE " + TOTAL_REGISTER_ENTRIES_TABLE + " SET COUNT=COUNT+1")
+    void increaseTotalEntriesInRegisterCount();
```

Comment:
discussed IRL -- apparently there may be a case where a postgres `SERIAL` type skips numbers if a transaction allocates a number and then rolls back.

Could we perhaps do this?

    create materialized view entry_count as select count(*) from ordered_entry_index;